### ISSUE TITLE
Image classification task

### ISSUE THREAD

**Issue by jemus42** [mlr-org/mlr3torch#1]
Task abstraction for image task, containing observation ID, path, label

- From folder with e.g. train/test folder, subfolders for classes
- From pre-built datasets from `{torchvision}`
- -> Ingest, spit out standardized task object

**Comment by jemus42** [comment]
Included example datasets in torch an keras tend to be binary formats or matrices, e.g.

```r
# torchvision dataset format, stores binary files on disk
cifar10_bin <- torchvision::cifar10_dataset(root = "/opt/example-data", download = TRUE)

# Keras version: Images in tensor
cifar10_keras <- keras::dataset_cifar10()
```

Handling these with the Task object might get tricky, so we're focusing on the ImageNet-like image folder structure first.
Helper `df_from_imagenet_dir` from mlr3keras is included verbatim for now and seems to work as expected using the [imagenette 160px dataset](https://github.com/fastai/imagenette) as test data for now:

``` r
library(mlr3torch)
df_from_imagenet_dir(c(
  "/opt/example-data/imagenette2-160/train/",
  "/opt/example-data/imagenette2-160/val/"
))
#>            class
#>     1: n01440764
#>     2: n01440764
#>     3: n01440764
#>     4: n01440764
#>     5: n01440764
#>    ---          
#> 13390: n03888257
#> 13391: n03888257
#> 13392: n03888257
#> 13393: n03888257
#> 13394: n03888257
#>                                                                                  image
#>     1: /opt/example-data/imagenette2-160/train//n01440764/ILSVRC2012_val_00000293.JPEG
#>     2: /opt/example-data/imagenette2-160/train//n01440764/ILSVRC2012_val_00002138.JPEG
#>     3: /opt/example-data/imagenette2-160/train//n01440764/ILSVRC2012_val_00003014.JPEG
#>     4: /opt/example-data/imagenette2-160/train//n01440764/ILSVRC2012_val_00006697.JPEG
#>     5: /opt/example-data/imagenette2-160/train//n01440764/ILSVRC2012_val_00007197.JPEG
#>    ---                                                                                
#> 13390:            /opt/example-data/imagenette2-160/val//n03888257/n03888257_9671.JPEG
#> 13391:            /opt/example-data/imagenette2-160/val//n03888257/n03888257_9770.JPEG
#> 13392:            /opt/example-data/imagenette2-160/val//n03888257/n03888257_9771.JPEG
#> 13393:            /opt/example-data/imagenette2-160/val//n03888257/n03888257_9812.JPEG
#> 13394:            /opt/example-data/imagenette2-160/val//n03888257/n03888257_9932.JPEG
```

<sup>Created on 2021-12-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Since this function combines train- and validation set it might be worth thinking about a helper down the road to restore the original train/val split to more easily compare results, maybe?

**Comment by pfistfl** [comment]
You can perhaps annotate rows with train and Test in the data.table but disable it as a feature when building the task ?

**Comment by sebffischer** [comment]
solved more generically with lazy_tensor
